src/binaryen-shell.cpp

```diff
@@ -275,20 +272,22 @@ static void run_asserts(size_t* i, bool* checked, AllocatingModule* wasm,
     if (id == ASSERT_INVALID) {
       // a module invalidity test
       AllocatingModule wasm;
-      bool invalid = false;
-      jmp_buf trapState;
-      if (setjmp(trapState) == 0) {
-        *builder = std::unique_ptr<SExpressionWasmBuilder>(new SExpressionWasmBuilder(wasm, *curr[1], [&]() {
-          invalid = true;
-          longjmp(trapState, 1);
-        }));
-      }
       if (print_before || print_after) {
         Colors::bold(std::cout);
         std::cerr << "printing in module invalidity test:\n";
         Colors::normal(std::cout);
         std::cout << wasm;
       }
+      bool invalid = false;
+      try {
+        *builder = std::unique_ptr<SExpressionWasmBuilder>(
+            new SExpressionWasmBuilder(wasm, *curr[1], [&]() {
+              invalid = true;
+              throw TrapException();
+            }));
+      } catch (const TrapException& e) {
+        invalid = true;
```

Comment:
and I guess I added this line and it's not strictly necessary. but maybe nice for clarity, better than the setting 4 lines above?